### PR TITLE
Add diagnostics to report when a folder referenced as a project in `azure.yaml` does not exist

### PR DESCRIPTION
**Pull request by bwateratmsft**
Part of #1388

How it looks in the `azure.yaml` file:
image

How it looks in the Problems window:
image

/cc @gkulin @savannahostrowski

### CODE REVIEW

File: ext/vscode/src/language/AzureYamlDiagnosticProvider.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                        }
+                    } catch {
+                        // Suppress the error--we'll emit our diagnostic below
+                    }
+                }
+
+                // If not, then emit an error diagnostic about it
+                results.push(new vscode.Diagnostic(
+                    new vscode.Range(document.positionAt(projectNode.range?.[0] ?? 0), document.positionAt(projectNode.range?.[1] ?? 0)),
+                    vscode.l10n.t('The project path must be an existing folder path relative to the azure.yaml file.'),
```
Comment (bwateratmsft) [inline ext/vscode/src/language/AzureYamlDiagnosticProvider.ts:69]: @gkulin are you happy with this error message string?

File: ext/vscode/src/language/AzureYamlDiagnosticProvider.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                            continue;
+                        }
+                    } catch {
+                        // Suppress the error--we'll emit our diagnostic below
+                    }
+                }
+
+                // If not, then emit an error diagnostic about it
+                results.push(new vscode.Diagnostic(
+                    new vscode.Range(document.positionAt(projectNode.range?.[0] ?? 0), document.positionAt(projectNode.range?.[1] ?? 0)),
```
Comment (karolz-ms) [inline ext/vscode/src/language/AzureYamlDiagnosticProvider.ts:68]: (minor nit) Consider assigning the `Range` to a separate variable--might be useful for debugging.

**Review by karolz-ms — APPROVED** [review]
Love it!

File: ext/vscode/src/language/AzureYamlDiagnosticProvider.ts (hunk 2 above)
Comment (bwateratmsft) [reply ext/vscode/src/language/AzureYamlDiagnosticProvider.ts:68]: I think I did exactly that at some point during debugging, so good idea :smile:

**Review by savannahostrowski — APPROVED** [review]
LGTM from my perspective. This will be so helpful!

@gkulin for final sign-off etc.

**Review by gkulin — APPROVED** [review]

**Review by ellismg — APPROVED** [review]

**Pull request merged**